packages/cli/src/lib/config/config.ts

```diff
@@ -44,9 +40,18 @@ export class Config {
       if (content instanceof Error) {
         throw content;
       }
+      if (!this.isSettingVersionInRange(content)) {
+        throw new IncompatibleConfiguration();
+      }
       return content;
     } catch (e) {
-      this.error(`Error while reading configuration at ${this.configPath}`);
+      if (e instanceof IncompatibleConfiguration) {
+        this.error(
+          `The configuration at ${this.configPath} is not compatible with this version of the CLI`
```

Comment:
Maybe we could explain "why" its not compatible ? The `IncompatibleConfiguration` error could contain a message that explains it.

Something like "no version found", "mismatch on major version", "incompatible format", etc.

Reply:
- [ ] Better logging